ansible/roles/test/tasks/snmp/memory.yml

```diff
@@ -0,0 +1,105 @@
+# Test SNMP Memory Utilization
+#
+#  - Check the difference percentage of Total Memory between SNMP and shell is 0
+#  - Check the difference percentage of Total Free Memory between SNMP and shell small than 3% when:
+#    * Check free memory before stress test
+#    * Check free memory in stress test
+#    * Check free memory after stress test
+#  - Check memory leak issue, the difference percentage between Total Free Memory before stress test and after should small than 3%
+#
+# Requires:  Ansible v2.0.0.2-1
+#
+# Usage:
+#
+#  ansible-playbook test_sonic.yml -i inventory --limit switch2 --become --tags snmp_memory -vvvvvv
+#
+
+- block:
+    - name: Copy Memory stress script to the switch
+      copy: src={{ item.src }} dest={{ item.dest }}
+      with_items:
+        - { src: 'roles/test/tasks/snmp/memory.py', dest: '/tmp/memory.py' }
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
+
+    - name: Check total memory via shell
+      shell: grep MemTotal /proc/meminfo | awk '{print $2}'
+      register: shell_total_memory
+      become: yes
+
+    - name: 'Validating SNMP total memory matches shell "/proc/meminfo" result'
+      assert:
+        that: "{{ (ansible_sysTotalMemery|int - shell_total_memory.stdout|int)|abs }} == 0"
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
```

Comment:
we already get one, do not need get again likely.

Reply:
Sure, will remove it.